yt_dlp/extractor/xiaohongshu.py

```diff
@@ -10,7 +10,7 @@
 
 
 class XiaoHongShuIE(InfoExtractor):
-    _VALID_URL = r'https?://www\.xiaohongshu\.com/explore/(?P<id>[\da-f]+)'
+    _VALID_URL = r'https?://www\.xiaohongshu\.com/(explore|discovery/item)/(?P<id>[\da-f]+)'
```

Comment:
[Don't capture groups you don't use](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#dont-capture-groups-you-dont-use)
```suggestion
    _VALID_URL = r'https?://www\.xiaohongshu\.com/(?:explore|discovery/item)/(?P<id>[\da-f]+)'
```